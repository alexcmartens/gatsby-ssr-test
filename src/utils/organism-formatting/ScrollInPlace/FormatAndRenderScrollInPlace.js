import React, { useEffect, useState } from "react"
import renderPaddingOrFullBleed from "../renderPaddingOrFullBleed"
import "./ScrollInPlace.scss"
import FormatAndRenderCtaWithHeader from "../CTA/FormatAndRenderCtaWithHeader"
import FormatAndRenderDataPoint from "../DataPoint/FormatAndRenderDataPoint"
import topLeftBorder from "../../../images/border-top-left.svg"
import topRightBorder from "../../../images/border-top-right.svg"
import bottomLeftBorder from "../../../images/border-bottom-left.svg"
import bottomRightBorder from "../../../images/border-bottom-right.svg"
import scrollIcon from "../../../images/scroll-icon.svg"
import useMouse from "../../hooks/useMouse"
import { ScrollInPlace } from "terra-one"
import HeaderConnector from "../../../components/Header/HeaderConnector"
import { LottieAnimation } from "terra-one"
import formatLottieAnimation from "../formatLottieAnimation"

const SlideContent = ({ LeftContent, CenterContent, RightContent }) => {
  return (
    <>
      <div className="content-sections desktop">
        <div className="left-section">
          <LeftContent />
        </div>

        <div className="center-section">
          <CenterContent />
        </div>

        <div className="right-section">
          <RightContent />
        </div>
      </div>

      <div className="content-sections tablet">
        <div className="sections-wrapper">
          <div className="left-section">
            <div className="top">
              <CenterContent />
            </div>
            <div className="bottom">
              <LeftContent />
            </div>
          </div>

          <div className="right-section">
            <RightContent />
          </div>
        </div>
      </div>

      <div className="content-sections mobile">
        <div className="sections-wrapper">
          <div className="background-section">
            <CenterContent />
          </div>

          <div className="scrollable-section">
            <div className="section background bg-dark-gray padded">
              <LeftContent />
            </div>

            <div className="section">
              <RightContent />
            </div>
          </div>
        </div>
      </div>
    </>
  )
}

const LeftContentComponent = ({ data }) => {
  if (data?.__typename === "ContentfulCtaWithTextAndHeader") {
    return (
      <FormatAndRenderCtaWithHeader
        data={data}
        nested={true}
        disablePadding={true}
      />
    )
  }
  return null
}

const CenterContentComponent = ({ data, imageVariant }) => {
  if (data?.__typename === "ContentfulImageWithFocalPoint") {
    return (
      <div
        className={`content-image-container ${
          imageVariant === "circle" ? "circle" : ""
        }`}
      >
        <img src={data?.image?.file?.url} alt={data?.altText} />
      </div>
    )
  } else if (data?.__typename === "ContentfulLottieAnimation") {
    const centerImageStyles = {}
    const centerComponent = formatLottieAnimation(
      data?.srcLink || data?.json?.data,
      centerImageStyles
    )
    return (
      <div
        className={`content-image-container ${
          imageVariant === "circle" ? "circle" : ""
        }`}
      >
        <LottieAnimation
          src={centerComponent?.src}
          style={centerComponent?.style}
          ratio={{ width: 1, height: 1 }}
        />
      </div>
    )
  }
  return null
}

const RightContentComponent = ({ data }) => {
  if (data?.__typename === "ContentfulCtaWithTextAndHeader") {
    return (
      <div className="background padded">
        <FormatAndRenderCtaWithHeader
          data={data}
          nested={true}
          disablePadding={true}
        />
      </div>
    )
  } else if (data?.__typename === "ContentfulDataPoint") {
    return (
      <div className="background bg-dark-gray">
        <FormatAndRenderDataPoint
          data={data}
          disablePadding={true}
          overrideTheme="transparent-light-text"
          overrideVariant="variant_2b"
        />
      </div>
    )
  }
  return null
}

const TitleContent = ({ data }) => {
  if (data?.__typename === "ContentfulHeader") {
    return (
      <div className="title-slide-content">
        <div className="title-content">
          <div className="border">
            <img src={topLeftBorder} alt="" />
            <img src={topRightBorder} alt="" />
          </div>
          <div className="title-inner">
            <HeaderConnector data={data} disablePadding={true} />
          </div>
          <div className="border">
            <img src={bottomLeftBorder} alt="" />
            <img src={bottomRightBorder} alt="" />
          </div>
        </div>
        <div className="scroll-icon">
          <img src={scrollIcon} alt="" />
        </div>
      </div>
    )
  }

  return null
}

const FormatAndRenderScrollInPlace = ({ data }) => {
  const [a11yModeEnabled, setA11yModeEnabled] = useState(false)
  const { mouse } = useMouse()

  useEffect(() => {
    mouse ? setA11yModeEnabled(false) : setA11yModeEnabled(true)
  }, [a11yModeEnabled, mouse])

  const titleBackgroundStyles = {}
  let titleBackground
  //Build the title slide
  if (data?.titleSlideBackground?.__typename === "ContentfulLottieAnimation") {
    titleBackground = {
      image: formatLottieAnimation(
        data?.titleSlideBackground?.srcLink ||
          data?.titleSlideBackground?.json?.data,
        titleBackgroundStyles
      ),
    }
  } else {
    titleBackground = {
      image: data?.titleSlideBackground?.image?.file?.url
        ? {
            src: data?.titleSlideBackground?.image?.file?.url,
            alt: data?.titleSlideBackground?.altText,
          }
        : null,
      video: data?.backgroundVideo?.file?.url,
    }
  }

  // Build the content slides
  const slides = []

  for (let i = 0; i < data?.slides?.length; i++) {
    const current = data?.slides?.[i]

    const slideHold = {
      backgroundImage: {
        src:
          current?.backgroundImage?.image?.file?.url ||
          "https://picsum.photos/id/152/1280/752",
        alt: current?.backgroundImage?.altText,
      },
      content: () => (
        <SlideContent
          LeftContent={() => (
            <LeftContentComponent data={current?.leftContent} />
          )}
          CenterContent={() => (
            <CenterContentComponent
              data={current?.centerImage}
              imageVariant={current?.centerImageVariant}
            />
          )}
          RightContent={() => (
            <RightContentComponent data={current?.rightContent} />
          )}
        />
      ),
    }

    slides.push(slideHold)
  }

  return renderPaddingOrFullBleed(
    <div className="scroll-in-place">
      <ScrollInPlace
        TitleContent={() => <TitleContent data={data?.titleSlideText} />}
        titleBackground={titleBackground}
        slides={slides}
        a11yMode={a11yModeEnabled}
      />
    </div>,
    a11yModeEnabled ? true : false,
    true,
    true,
    false
  )
}

export default FormatAndRenderScrollInPlace
