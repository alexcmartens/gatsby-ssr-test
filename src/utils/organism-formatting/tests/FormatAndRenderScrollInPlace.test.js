import React from "react"
import renderer from "react-test-renderer"
import FormatAndRenderScrollInPlace from "../ScrollInPlace/FormatAndRenderScrollInPlace"

describe.skip("FormatAndRenderScrollInPlace", () => {
  it("renders correctly", () => {
    const data = {
      __typename: "ContentfulScrollInPlace",
      titleSlideBackground: {
        id: "84c6e229-53b0-50a0-a328-e29ead42839d",
        focalPoint: {
          focalPoint: {
            x: 537,
            y: 177,
          },
        },
        image: {
          title: "Front Frog",
          file: {
            url: "//images.ctfassets.net/citn2sn5tdjr/5FRNqx8tTG9ewzghjTD0ft/a59add5717655ebcb357dbae7b31104b/frog-897418_1920.jpg",
            details: {
              image: {
                height: 1276,
                width: 1920,
              },
            },
          },
        },
        __typename: "ContentfulImageWithFocalPoint",
        title: "JJung Test Pull Quote Side Image",
        altText: "placeholder",
      },
      backgroundVideo: {
        file: {
          url: "//videos.ctfassets.net/citn2sn5tdjr/7hO7fIlKpIZrXYSABmdzU9/62056977997e3ce3327fe9ff7499ed18/video_preview_h264.mp4",
        },
      },
      titleSlideText: {
        __typename: "ContentfulHeader",
        colorScheme: null,
        eyebrow: "eyebrow",
        headline: "headline",
        subhead: "subhead",
        jumpNavTitle: "jump nav title",
        anchor: "/",
      },
      slides: [
        {
          __typename: "ContentfulSlide",
          backgroundImage: {
            id: "b97b8fd6-d341-5fcf-a381-ff2e16149d73",
            focalPoint: {
              focalPoint: {
                x: 3463,
                y: 2017,
              },
            },
            image: {
              title: "dog",
              file: {
                url: "//images.ctfassets.net/citn2sn5tdjr/7wBhLKP8AN67gmNO0MJzsP/8da854bf7a41a2f2b703cbe5d1d2ae2b/dog.jpg",
                details: {
                  image: {
                    height: 4000,
                    width: 6000,
                  },
                },
              },
            },
            __typename: "ContentfulImageWithFocalPoint",
            title: "New Test dog image",
            altText: "alt text",
          },
          leftContent: {
            __typename: "ContentfulCtaWithTextAndHeader",
            variant: "dark",
            alignment: "left",
            eyebrow: "test",
            headline: "test",
            subhead: "test",
            jumpNavTitle: "test",
            anchor: "test",
            body: {
              body: "test",
              childMarkdownRemark: {
                html: "<p>test</p>",
              },
            },
            displayButtonsAsLinks: true,
            ctaButtons: [
              {
                __typename: "ContentfulExternalLink",
                id: "4d0c2fc5-b317-5d63-b5b3-523e40041904",
                url: "https://www.google.com/",
                text: "DO NOT DELETE - placeholder external link",
              },
              {
                __typename: "ContentfulExternalLink",
                id: "4d0c2fc5-b317-5d63-b5b3-523e40041904",
                url: "https://www.google.com/",
                text: "DO NOT DELETE - placeholder external link",
              },
            ],
          },
          centerImage: {
            id: "0a32ef0a-b428-5329-bf28-78e80606a57a",
            focalPoint: {
              focalPoint: null,
            },
            image: {
              title:
                "industries-overview-classifying-caribbean-coral-272x180.jpg",
              file: {
                url: "//images.ctfassets.net/citn2sn5tdjr/2V4FylU8tbTOXy7TopNjMS/f6b0a4227827dc08d94a04266c1712da/industries-overview-classifying-caribbean-coral-272x180.jpg",
                details: {
                  image: {
                    height: 180,
                    width: 272,
                  },
                },
              },
            },
            __typename: "ContentfulImageWithFocalPoint",
            title: "Classifying the Caribbean's coral image",
            altText: null,
          },
          rightContent: {
            __typename: "ContentfulCtaWithTextAndHeader",
            variant: "dark",
            alignment: "left",
            eyebrow: null,
            headline: "Climate action",
            subhead: null,
            jumpNavTitle: null,
            anchor: null,
            body: {
              body: 'Trimble is acting with urgency to combat the negative impacts of climate change by reducing our carbon footprint and by providing innovative solutions to enable our customers to do the same. We are also setting [science-based targets in line with the commitment](https://www.google.com/ "test link") to keep global warming to no more than 1.5 degrees celsius.',
              childMarkdownRemark: {
                html: '<p>Trimble is acting with urgency to combat the negative impacts of climate change by reducing our carbon footprint and by providing innovative solutions to enable our customers to do the same. We are also setting <a href="https://www.google.com/" title="test link">science-based targets in line with the commitment</a> to keep global warming to no more than 1.5 degrees celsius.</p>',
              },
            },
            displayButtonsAsLinks: true,
            ctaButtons: null,
          },
          centerImageVariant: "square",
        },
        {
          __typename: "ContentfulSlide",
          backgroundImage: {
            id: "b97b8fd6-d341-5fcf-a381-ff2e16149d73",
            focalPoint: {
              focalPoint: {
                x: 3463,
                y: 2017,
              },
            },
            image: {
              title: "dog",
              file: {
                url: "//images.ctfassets.net/citn2sn5tdjr/7wBhLKP8AN67gmNO0MJzsP/8da854bf7a41a2f2b703cbe5d1d2ae2b/dog.jpg",
                details: {
                  image: {
                    height: 4000,
                    width: 6000,
                  },
                },
              },
            },
            __typename: "ContentfulImageWithFocalPoint",
            title: "New Test dog image",
            altText: "alt text",
          },
          leftContent: {
            __typename: "ContentfulCtaWithTextAndHeader",
            variant: "dark",
            alignment: "left",
            eyebrow: "test",
            headline: "test",
            subhead: "test",
            jumpNavTitle: "test",
            anchor: "test",
            body: {
              body: "test",
              childMarkdownRemark: {
                html: "<p>test</p>",
              },
            },
            displayButtonsAsLinks: true,
            ctaButtons: [
              {
                __typename: "ContentfulExternalLink",
                id: "4d0c2fc5-b317-5d63-b5b3-523e40041904",
                url: "https://www.google.com/",
                text: "DO NOT DELETE - placeholder external link",
              },
              {
                __typename: "ContentfulExternalLink",
                id: "4d0c2fc5-b317-5d63-b5b3-523e40041904",
                url: "https://www.google.com/",
                text: "DO NOT DELETE - placeholder external link",
              },
            ],
          },
          centerImage: {
            id: "0a32ef0a-b428-5329-bf28-78e80606a57a",
            focalPoint: {
              focalPoint: null,
            },
            image: {
              title:
                "industries-overview-classifying-caribbean-coral-272x180.jpg",
              file: {
                url: "//images.ctfassets.net/citn2sn5tdjr/2V4FylU8tbTOXy7TopNjMS/f6b0a4227827dc08d94a04266c1712da/industries-overview-classifying-caribbean-coral-272x180.jpg",
                details: {
                  image: {
                    height: 180,
                    width: 272,
                  },
                },
              },
            },
            __typename: "ContentfulImageWithFocalPoint",
            title: "Classifying the Caribbean's coral image",
            altText: null,
          },
          rightContent: {
            __typename: "ContentfulDataPoint",
            colorScheme: "Trimble Blue",
            variant: "Horizontal",
            title: "DO NOT DELETE - placeholder data point",
            jumpNavTitle: "jump nav title",
            anchor: "/",
            points: [
              {
                description: "test",
                header: "DO NOT DELETE - placeholder point",
                icon: "Enclosed Arrow Up",
              },
              {
                description: "test",
                header: "DO NOT DELETE - placeholder point",
                icon: "Enclosed Arrow Up",
              },
              {
                description: "test",
                header: "DO NOT DELETE - placeholder point",
                icon: "Enclosed Arrow Up",
              },
            ],
            theme: "transparent-light-text",
          },
          centerImageVariant: "circle",
        },
      ],
    }

    const tree = renderer
      .create(<FormatAndRenderScrollInPlace data={data} />)
      .toJSON()
    expect(tree).toMatchSnapshot()
  })
})
