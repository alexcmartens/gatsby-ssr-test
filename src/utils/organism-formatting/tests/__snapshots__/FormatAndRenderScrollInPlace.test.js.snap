// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FormatAndRenderScrollInPlace renders correctly 1`] = `
<div
  className="scroll-in-place"
>
  <div
    TitleContent={[Function]}
    slides={
      Array [
        Object {
          "backgroundImage": Object {
            "alt": "alt text",
            "src": "//images.ctfassets.net/citn2sn5tdjr/7wBhLKP8AN67gmNO0MJzsP/8da854bf7a41a2f2b703cbe5d1d2ae2b/dog.jpg",
          },
          "content": [Function],
        },
        Object {
          "backgroundImage": Object {
            "alt": "alt text",
            "src": "//images.ctfassets.net/citn2sn5tdjr/7wBhLKP8AN67gmNO0MJzsP/8da854bf7a41a2f2b703cbe5d1d2ae2b/dog.jpg",
          },
          "content": [Function],
        },
      ]
    }
    titleBackground={
      Object {
        "image": Object {
          "alt": "placeholder",
          "src": "//images.ctfassets.net/citn2sn5tdjr/5FRNqx8tTG9ewzghjTD0ft/a59add5717655ebcb357dbae7b31104b/frog-897418_1920.jpg",
        },
        "video": "//videos.ctfassets.net/citn2sn5tdjr/7hO7fIlKpIZrXYSABmdzU9/62056977997e3ce3327fe9ff7499ed18/video_preview_h264.mp4",
      }
    }
  />
</div>
`;
